i2c/simulation/i2c_sim.cpp

```diff
@@ -14,12 +14,20 @@ auto SimI2C::central_transmit(uint8_t *data, uint16_t size,
     std::copy_n(data, size, last_transmitted.begin());
     transmit_count++;
     if (!sensor_map.empty()) {
-        uint16_t store_in_register = 0;
-        auto *iter = data + 1;
-        iter = bit_utils::bytes_to_int(iter, data + size, store_in_register);
-        LOG("Storing %d in register %d", store_in_register, reg);
-        sensor_map[dev_address].REGISTER_MAP[reg] = store_in_register;
         next_register_map[dev_address] = reg;
+        if (size > 1) {
+            uint16_t store_in_register = 0;
+            auto *iter = data + 1;
+            if (size == 3) {
+                iter = bit_utils::bytes_to_int(iter, data + size,
+                                               store_in_register);
+            } else if (size == 5) {
+                uint32_t store_in_register = 0;
```

Comment:
This definitely won't do what you think. It will save the value to the variable defined on line 25 and not the one defined on line 19, so you'll write 0 on line 29 always. I think you have to pull the assignment into the `if` clauses.